crates/paimon/src/spec/types.rs

```diff
@@ -223,6 +227,44 @@ impl Default for BinaryType {
     }
 }
 
+impl FromStr for BinaryType {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split_whitespace().collect();
```

Comment:
Hi, every `collect` here involves an allocation. It would be better if we could avoid it.

We can use `find` instead of `split` and `collect`, for example:

```rust
impl FromStr for BinaryType {
    type Err = Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        if !s.starts_with(serde_utils::BINARY::NAME) {
            return DataTypeInvalidSnafu {
                message: xxx
            }
            .fail();
        }

        let Some(open_bracket) = s.find('(') else {
             return DataTypeInvalidSnafu {
                message: xxx
            }
            .fail();
        };
        let Some(close_bracket) = s.find(')') else {
             return DataTypeInvalidSnafu {
                message: xxx
            }
            .fail();
        };

        if open_bracket >= close_bracket {
            return DataTypeInvalidSnafu {
                message: xxx
            }
            .fail();
        }

        let length_str = &s[open_bracket + 1..close_bracket];
        let length = length_str.parse::<usize>().map_err(|_| DataTypeInvalidSnafu {
            message: xxx
        })?;

        let nullable = !s[close_bracket..].contains("NOT NULL");

        Ok(BinaryType {
            nullable,
            length,
        })
    }
}

```